pkg/scheduler/algorithm/predicates/error.go

```diff
@@ -23,58 +23,57 @@ import (
 )
 
 var (
-	// The predicateName tries to be consistent as the predicate name used in DefaultAlgorithmProvider defined in
-	// defaults.go (which tend to be stable for backward compatibility)
-
 	// NOTE: If you add a new predicate failure error for a predicate that can never
 	// be made to pass by removing pods, or you change an existing predicate so that
 	// it can never be made to pass by removing pods, you need to add the predicate
 	// failure error in nodesWherePreemptionMightHelp() in scheduler/core/generic_scheduler.go
 
 	// ErrDiskConflict is used for NoDiskConflict predicate error.
-	ErrDiskConflict = newPredicateFailureError("NoDiskConflict", "node(s) had no available disk")
+	ErrDiskConflict = newPredicateFailureError(NoDiskConflictPred, "node(s) had no available disk")
 	// ErrVolumeZoneConflict is used for NoVolumeZoneConflict predicate error.
-	ErrVolumeZoneConflict = newPredicateFailureError("NoVolumeZoneConflict", "node(s) had no available volume zone")
+	ErrVolumeZoneConflict = newPredicateFailureError(NoVolumeZoneConflictPred, "node(s) had no available volume zone")
 	// ErrNodeSelectorNotMatch is used for MatchNodeSelector predicate error.
-	ErrNodeSelectorNotMatch = newPredicateFailureError("MatchNodeSelector", "node(s) didn't match node selector")
+	ErrNodeSelectorNotMatch = newPredicateFailureError(MatchNodeSelectorPred, "node(s) didn't match node selector")
 	// ErrPodAffinityNotMatch is used for MatchInterPodAffinity predicate error.
-	ErrPodAffinityNotMatch = newPredicateFailureError("MatchInterPodAffinity", "node(s) didn't match pod affinity/anti-affinity")
+	ErrPodAffinityNotMatch = newPredicateFailureError(MatchInterPodAffinityPred, "node(s) didn't match pod affinity/anti-affinity")
 	// ErrPodAffinityRulesNotMatch is used for PodAffinityRulesNotMatch predicate error.
-	ErrPodAffinityRulesNotMatch = newPredicateFailureError("PodAffinityRulesNotMatch", "node(s) didn't match pod affinity rules")
+	ErrPodAffinityRulesNotMatch = newPredicateFailureError(MatchInterPodAffinityPred, "node(s) didn't match pod affinity rules")
 	// ErrPodAntiAffinityRulesNotMatch is used for PodAntiAffinityRulesNotMatch predicate error.
-	ErrPodAntiAffinityRulesNotMatch = newPredicateFailureError("PodAntiAffinityRulesNotMatch", "node(s) didn't match pod anti-affinity rules")
+	ErrPodAntiAffinityRulesNotMatch = newPredicateFailureError(MatchInterPodAffinityPred, "node(s) didn't match pod anti-affinity rules")
```

Comment:
@resouer  maybe, we should change `re.Error()` to `re.GetReason()` in pkg/kubelet/lifecycle/predicate.go#L129-L136?   `re.Error()` only return the predicate name, howerver each predicate name may map to many diferent error, such as `MatchInterPodAffinityPred` above.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/lifecycle/predicate.go#L129-L136

Reply:
The kubelet side logic is not wrong.

While as you said, you need to keep `PodAffinityRulesNotMatch` and `PodAntiAffinityRulesNotMatch` as is, they're special cases. You can define constants for them separately in this file.

Reply:
okay, I'll define some constants.
another question is: to some degree, the first parameter of `newPredicateFailureError` is not always the predicate name and it seems more reasonable rename the struct `PredicateFailureError`'s member from `PredicateName` to `ErrorType`;

in other words, rename from
```golang
// PredicateFailureError describes a failure error of predicate.
type PredicateFailureError struct {
	PredicateName string
	PredicateDesc string
}
```
to
```golang
// PredicateFailureError describes a failure error of predicate.
type PredicateFailureError struct {
	ErrorType string
        ErrorDesc string
}
```